python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py

```diff
@@ -665,7 +665,8 @@ def resolve_assetNodes(self, graphene_info, **kwargs):
         ]
 
     def resolve_assetNodeOrError(self, graphene_info, **kwargs):
-        return get_asset_node(graphene_info, AssetKey.from_graphql_input(kwargs["assetKey"]))
+        asset_key_input = cast(Mapping[str, Sequence[str]], kwargs.get("assetKey"))
```

Comment:
Shouldn't we be doing a dynamic runtime check here to match the cast to ensure that the `cast` isn't lying?

Reply:
>Have slight preference here for factoring out the cast to a helper function that also does a runtime check, converts Sequence[AssetKey] and notes that this is always paired with a graphene field with type graphene.Argument(non_null_list(GrapheneAssetKeyInput))

This is addressed upstack in https://github.com/dagster-io/dagster/pull/11724

That PR eliminates all use of `**kwargs` in resolvers so no casts are needed anymore.